src/authenticationservices.cs

```diff
@@ -930,4 +951,342 @@ interface ASAccountAuthenticationModificationViewController {
 		void CancelRequest ();
 	}
 
+	[iOS (15,0), Mac (12,0), MacCatalyst (15,0), NoWatch, NoTV]
+	[Static]
+	interface ASAuthorizationPublicKeyCredentialAttestationKind
+	{
+		[Field ("ASAuthorizationPublicKeyCredentialAttestationKindNone")]
+		NSString None { get; }
+
+		[Field ("ASAuthorizationPublicKeyCredentialAttestationKindDirect")]
+		NSString Direct { get; }
+
+		[Field ("ASAuthorizationPublicKeyCredentialAttestationKindIndirect")]
+		NSString Indirect { get; }
+
+		[Field ("ASAuthorizationPublicKeyCredentialAttestationKindEnterprise")]
+		NSString Enterprise { get; }
+	}
+
+	[iOS (15,0), Mac (12,0), MacCatalyst (15,0), NoWatch, NoTV]
+	[Static]
+	interface ASAuthorizationPublicKeyCredentialUserVerificationPreference
+	{
+		[Field ("ASAuthorizationPublicKeyCredentialUserVerificationPreferencePreferred")]
+		NSString Preferred { get; }
+
+		[Field ("ASAuthorizationPublicKeyCredentialUserVerificationPreferenceRequired")]
+		NSString Required { get; }
+
+		[Field ("ASAuthorizationPublicKeyCredentialUserVerificationPreferenceDiscouraged")]
+		NSString Discouraged { get; }
+	}
+
+	[iOS (15,0), Mac (12,0), MacCatalyst (15,0), NoWatch, NoTV]
+	enum ASAuthorizationPublicKeyCredentialResidentKeyPreference {
+		[Field ("ASAuthorizationPublicKeyCredentialResidentKeyPreferenceDiscouraged")]
+		Discouraged,
+		[Field ("ASAuthorizationPublicKeyCredentialResidentKeyPreferencePreferred")]
+		Preferred,
+		[Field ("ASAuthorizationPublicKeyCredentialResidentKeyPreferenceRequired")]
+		Required,
+	}
+
+	[iOS (15,0), Mac (12,0), MacCatalyst (15,0), NoWatch, NoTV]
+	enum ASAuthorizationSecurityKeyPublicKeyCredentialDescriptorTransport {
+		[Field ("ASAuthorizationSecurityKeyPublicKeyCredentialDescriptorTransportUSB")]
+		Usb,
+		[Field ("ASAuthorizationSecurityKeyPublicKeyCredentialDescriptorTransportNFC")]
+		Nfc,
+		[Field ("ASAuthorizationSecurityKeyPublicKeyCredentialDescriptorTransportBluetooth")]
+		Bluetooth,
+	}
+
+	interface IASAuthorizationPublicKeyCredentialAssertion { }
+
+	[NoWatch, Mac (12,0), iOS (15,0), MacCatalyst (15,0), NoTV]
+	[Protocol]
+	interface ASAuthorizationPublicKeyCredentialAssertion : ASPublicKeyCredential
+	{
+		[Abstract]
+		[Export ("rawAuthenticatorData", ArgumentSemantic.Copy)]
+		NSData RawAuthenticatorData { get; }
+
+		[Abstract]
+		[Export ("userID", ArgumentSemantic.Copy)]
+		NSData UserId { get; }
+
+		[Abstract]
+		[Export ("signature", ArgumentSemantic.Copy)]
+		NSData Signature { get; }
+	}
+
+	[NoWatch, Mac (12,0), iOS (15,0), MacCatalyst (15,0), NoTV]
+	[BaseType (typeof (ASAuthorizationRequest))]
+	[DisableDefaultCtor]
+	interface ASAuthorizationPlatformPublicKeyCredentialAssertionRequest : ASAuthorizationPublicKeyCredentialAssertionRequest
+	{
+		[Sealed]
```

Comment:
why ? iow worth a comment

Reply:
So I actually don't who told me to add the Sealed, perhaps @dalexsoto ?
But when I remove it, I get these build errors: https://gist.github.com/tj-devel709/d459d9e717a186486deab6479f57c050

Reply:
That error is because you're using the same selector for multiple methods/properties.

Why is that called `PlatformAllowedCredentials` if the selector is `allowedCredentials` ?

Reply:
Oh I just remembered what is going on! Alex did help me with this one a few weeks ago!

So there are two different classes that use this similar "allowedCredentials" header: `ASAuthorizationPlatformPublicKeyCredentialAssertionRequest` and `ASAuthorizationSecurityKeyPublicKeyCredentialAssertionRequest`

The return type of the former is `ASAuthorizationPlatformPublicKeyCredentialDescriptor[]`
and the return type of the second one is `ASAuthorizationSecurityKeyPublicKeyCredentialDescriptor[]`

but they both are overriding a property from the protocol `ASAuthorizationPublicKeyCredentialAssertionRequest` 
which has the base property "allowedCredentials" that is of type `IASAuthorizationPublicKeyCredentialDescriptor[]`

So to get past this typing confusion, Alex suggested giving them different names and sealing the properties in the child classes. I definitely should have made a comment because I had already forgotten about this.

Reply:
That used to be a problem but C# now have covariant returns
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/covariant-returns

I wonder if _just_ using `[Override]` would work...
or if more change are needed

The main _problem_ with this rename-approach is that it's not easily discoverable by developers - so they might still use the original (base) one and have to deal with casts

Reply:
Using just the Override attribute gives me the following [errors](https://gist.github.com/tj-devel709/c4f0b6a35c8fe82332904237032e98e2)

I was wondering if maybe the protocol interface method needs a virtual attribute?
```
[The Sealed Attribute] Instructs the generator to flag the generated method as sealed. 
If this attribute is not specified, the default is to generate a virtual method 
(either a virtual method, an abstract method or an override depending on how other attributes are used).
```
This method is marked abstract and is maybe not virtual by default because of that?

Reply:
it's not a big deal, let's land this as-is (unless someone else objects) and I might have a have a quick look